akka-typed/src/main/scala/akka/typed/ActorContext.scala

```diff
@@ -186,7 +186,14 @@ class StubbedActorContext[T](
 
   def executionContext: ExecutionContextExecutor = system.executionContext
 
-  def spawnAdapter[U](f: U ⇒ T): ActorRef[U] = spawnAnonymous[Any](Behavior.emptyBehavior)
+  def spawnAdapter[U](f: U ⇒ T): ActorRef[U] = {
+    val i = Inbox[U](childName.next())
```

Comment:
They share the same counter, but that should not be a problem. The important difference is that the adapter is actually functional in that it will enqueue to the main inbox (otherwise the synchronous tests would not be possible).